### PR TITLE
fix: Build arm64 docker images with arm64 platform

### PR DESCRIPTION
**Pull request by disq**
Could fix https://github.com/cloudquery/cloudquery/issues/16218

https://goreleaser.com/cookbooks/multi-platform-docker-images/#creating-multi-platform-docker-images-with-goreleaser also mentions qemu, but I'm not sure it's needed after checking https://github.com/docker/setup-qemu-action/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

### CODE REVIEW

**Review by erezrokah — COMMENTED** [review]
I think you'll need to add:

```yaml
-
  name: Set up Docker Buildx
  uses: docker/setup-buildx-action@v3
```

Before https://github.com/cloudquery/cloudquery/blob/9379d5348bdc3abae8bad2f426eb7a513e7b2a83/.github/workflows/release_cli.yml#L43

File: cli/.goreleaser.yaml [hunk 1]
```diff
@@ -32,7 +35,10 @@ dockers:
       - "ghcr.io/cloudquery/cloudquery:latest-linux-arm64"
       - "ghcr.io/cloudquery/cloudquery:{{.Version}}-linux-arm64"
       - "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}-linux-arm64"
+    use: buildx
     build_flag_templates:
+      - "--pull"
```
Comment (hermanschaaf) [inline cli/.goreleaser.yaml:40]: Does buildx not account for the platform when checking if the image is already in the cache, or what is the reason for adding `--pull` ?

**Review by erezrokah — COMMENTED** [review]
We might want to follow up with a manifest configuration, as currently we tag `amd64` as:

```
- "ghcr.io/cloudquery/cloudquery:latest"
- "ghcr.io/cloudquery/cloudquery:{{.Version}}"
- "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}"
```

**Comment by erezrokah** [comment]
Also probably you could test this via `goreleaser release --snapshot --clean -f ./cli/.goreleaser.yaml` (you need the pro version from `brew install goreleaser/tap/goreleaser-pro` (API key in 1Password)

File: cli/.goreleaser.yaml (hunk 1 above)
Comment (disq) [reply cli/.goreleaser.yaml:40]: I've added it as per Erez's [instruction](https://github.com/cloudquery/cloudquery/issues/16218#issuecomment-1903804664) feel free to take over this issue if you're comfortable with this as I'm really not.

Comment (hermanschaaf) [reply cli/.goreleaser.yaml:40]: Oh I see, it's from the goreleaser example. I guess it doesn't hurt, we just won't use the cache. I also don't know whether that's necessary 😅

Comment (erezrokah) [reply cli/.goreleaser.yaml:40]: I think the reason to add `--pull` is to refresh any base images our image depends on. For example if we depend on `v2` tag of an image we'll pull the latest changes to that tag.

Comment (hermanschaaf) [reply cli/.goreleaser.yaml:40]: Ah I see, makes sense yeah.

**Comment by disq** [comment]
@erezrokah added (not sure correctly?) in [`f2e31af` (#16231)](https://github.com/cloudquery/cloudquery/pull/16231/commits/f2e31af9deb5d3e9d9fd5a5fa7b2327e4eaece75) PTAL

goreleaser seems to work (once I switched to docker default context)

File: cli/.goreleaser.yaml [hunk 2]
```diff
@@ -32,9 +35,28 @@ dockers:
...
+
+- name_template: "cloudquery/cloudquery:latest"
+  image_templates:
+  - "cloudquery/cloudquery:latest-linux-amd64"
+  - "cloudquery/cloudquery:latest-linux-arm64"
+
+- name_template: "cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}"
+  image_templates:
+  - "cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}-linux-amd64"
+  - "cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}-linux-arm64"
```
Comment (erezrokah) [inline cli/.goreleaser.yaml:58]:
```suggestion
- name_template: "cloudquery/cloudquery:{{.Version}}"
  image_templates:
  - "ghcr.io/cloudquery/cloudquery:{{.Version}}-linux-amd64"
  - "ghcr.io/cloudquery/cloudquery:{{.Version}}-linux-arm64"

- name_template: "ghcr.io/cloudquery/cloudquery:latest"
  image_templates:
  - "ghcr.io/cloudquery/cloudquery:latest-linux-amd64"
  - "ghcr.io/cloudquery/cloudquery:latest-linux-arm64"

- name_template: "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}"
  image_templates:
  - "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}-linux-amd64"
  - "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}-linux-arm64"
```

And I think you need to remove:

```
      - "ghcr.io/cloudquery/cloudquery:latest"
      - "ghcr.io/cloudquery/cloudquery:{{.Version}}"
      - "ghcr.io/cloudquery/cloudquery:{{ .Major }}.{{ .Minor }}"
```

from the amd64 configuration.

**Comment by disq** [comment]
@erezrokah Makes much more sense now thanks

**Review by erezrokah — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**